source/extensions/filters/http/cache/http_cache.cc

```diff
@@ -0,0 +1,138 @@
+#include "extensions/filters/http/cache/http_cache.h"
+
+#include <algorithm>
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/cache/http_cache_utils.h"
+
+#include "absl/time/time.h"
+
+using absl::Duration;
+using absl::FromChrono;
+using absl::ZeroDuration;
+using Envoy::Extensions::HttpFilters::Cache::Internal::effectiveMaxAge;
+using Envoy::Extensions::HttpFilters::Cache::Internal::httpTime;
+using Envoy::Http::HeaderEntry;
+using Envoy::Http::HeaderMap;
+using Envoy::Http::HeaderMapPtr;
+using Envoy::Http::HeaderString;
+using Envoy::Http::LowerCaseString;
+using std::string;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+LookupRequest::LookupRequest(const HeaderMap& request_headers, SystemTime timestamp)
+    : timestamp_(timestamp),
+      request_cache_control_(request_headers.CacheControl() == nullptr
+                                 ? ""
+                                 : request_headers.CacheControl()->value().getStringView()) {
+  RELEASE_ASSERT(request_headers.Path(),
+                 "Can't form cache lookup key for malformed HeaderMap with null Path.");
```

Comment:
is this reachable via the API? Consider generating an error response rather than asserting here; I"m thinking about how noisy this may make fuzzers, and also possibly represent something reachable externally as a DoS.

Could do an ASSERT though maybe.

Reply:
A buggy creator of LookupRequests could cause this condition. The only non-test creator of LookupRequests is CacheFilter; I added checks to (in CacheFilter::isCacheable) to reject such malformed headers, and it's currently the only creator of LookupRequests. I changed the RELEASE_ASSERTs to ASSERTs, as the style guide says to prefer null pointer dereferences over RELEASE_ASSERTs.